### PR TITLE
feat(rproxy): customizable SSO auth shield

### PR DESCRIPTION
**Pull request by deshmukhmayur**
# Closes

SSO Authentication Shield in the reverse proxy

# Explain the feature/fix

- Added dynamic SSO checks for application routes
- Added `.prettierrc` for enforcing coding standards

## Does this PR introduce a breaking change

No.

## Screenshots

N/A

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demoed and the design review approved?

### CODE REVIEW

File: packages/reverse-proxy-service/src/couchdb/resolver.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { createHmac } from "crypto";
 import { Request, Response, NextFunction } from "express";
 import { createProxyMiddleware } from "http-proxy-middleware";
-import { COUCHDB_HOST, COUCHDB_SECRET } from "../config/env";
+import { COUCHDB_HOST, COUCHDB_SECRET } from "../setup/env";
```
Comment (riginoommen) [inline packages/reverse-proxy-service/src/couchdb/resolver.ts:4]: Single quotes should be used

File: packages/reverse-proxy-service/src/index.ts [hunk 2]
```diff
@@ -1,7 +1,20 @@
-import app from './app';
-import { PORT } from './config/env';
-import logger from './utils/logger';
+import getServer from "./server";
```
Comment (riginoommen) [inline packages/reverse-proxy-service/src/index.ts:1]: Single quotes should be used

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: packages/reverse-proxy-service/src/couchdb/resolver.ts (hunk 1 above)
Comment (deshmukhmayur) [reply packages/reverse-proxy-service/src/couchdb/resolver.ts]: Fixed. Added a `.prettierrc` to overwrite the defaults.

File: packages/reverse-proxy-service/src/index.ts (hunk 2 above)
Comment (deshmukhmayur) [reply packages/reverse-proxy-service/src/index.ts]: Fixed.

**Review by riginoommen — APPROVED** [review]

**Pull request merged**